flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSerializer.java

```diff
@@ -79,15 +87,16 @@
 
 	// -------- runtime fields, non-serializable, lazily initialized -----------
 
-	private transient SpecificDatumWriter<T> writer;
-	private transient SpecificDatumReader<T> reader;
+	private transient GenericDatumWriter<T> writer;
+	private transient GenericDatumReader<T> reader;
 
 	private transient DataOutputEncoder encoder;
 	private transient DataInputDecoder decoder;
 
-	private transient SpecificData avroData;
+	private transient GenericData avroData;
 
 	private transient Schema schema;
+	private final String schemaString;
```

Comment:
As per the comments, the existing code orders config fields before runtime fields. Can you place the schema to match that pattern?